### ISSUE TITLE
Clarify what "data values" are in the docs

### ISSUE THREAD

**Issue by jayunit100** [k14s/ytt#480]
The doc page for https://carvel.dev/ytt/docs/latest/ytt-data-values/ is self referential :)
Also the https://carvel.dev/ytt/docs/latest/how-to-use-data-values/ page seems to reference "Data Values" as a proper qouted term, but doesnt seem to really explain it either.

```
Overview ¶
A ytt run can be configured by supplying custom Data Values.
```

Can we have `Data Values`  definition on that page, where we sorta dumb it down... i.e. 
It can be a little confusing if you dont use YTT alot, to try to differentiate the data, template, and value...

**Comment by pivotaljohn** [comment]
_(originated out of this conversation: https://kubernetes.slack.com/archives/CH8KCCKA5/p1630019986016400)_

**Comment by pivotaljohn** [comment]
@jayunit100 here's a sketch: [How to Use Data Values](https://github.com/vmware-tanzu/carvel/blob/1a229852433ba674199ad7a3e7acaa3b744caa12/site/content/ytt/docs/latest/how-to-use-data-values.md) ([diff](https://github.com/vmware-tanzu/carvel/pull/227/files?short_path=5fb7904#diff-5fb7904cecea924fb7c71a5af93f6a3079e6947dbcb6864748ccc71159f880ae))
 
In what ways could we make it even clearer?

**Comment by pivotaljohn** [comment]
Closing this for now: we've made an improvement. Of course, there's always room for more.

@jayunit100, if you have more ideas along these lines, feel free to either comment on this issue (we'll re-open) or create a new issue.

Thanks for your support. 👍🏻
